crates/exex/src/hints.rs

```diff
@@ -0,0 +1,93 @@
+use crate::{db::Database, exex::DATABASE_PATH};
+use cairo_vm::{
+    hint_processor::{
+        builtin_hint_processor::builtin_hint_processor_definition::{
+            BuiltinHintProcessor, HintFunc,
+        },
+        hint_processor_definition::HintReference,
+    },
+    serde::deserialize_program::ApTracking,
+    types::exec_scope::ExecutionScopes,
+    vm::{errors::hint_errors::HintError, vm_core::VirtualMachine},
+    Felt252,
+};
+use rusqlite::Connection;
+use std::{collections::HashMap, fmt, rc::Rc};
+
+/// A generic structure to encapsulate a hint with a closure that contains the specific logic.
+pub struct Hint {
+    /// The name of the hint.
+    name: String,
+    /// The function containing the hint logic.
+    func: Rc<HintFunc>,
+}
+
+impl fmt::Debug for Hint {
```

Comment:
rust noob question, what's the diff between this and deriving `Debug` ?

Reply:
Here we can't derive as `Rc` doesn't implement `Debug`...